composer.json

```diff
@@ -15,17 +15,26 @@
   },
   "require": {
     "php": ">=8.0",
-    "ext-curl": "*",
     "ext-json": "*",
-    "psr/log": "~1.0 || ~2.0 || ~3.0"
+    "psr/log": "~1.0 || ~2.0 || ~3.0",
+    "psr/http-client": "^1.0",
+    "psr/http-client-implementation": "^1.0",
+    "guzzlehttp/psr7": "^2.4",
+    "php-http/discovery": "^1.15"
   },
   "require-dev": {
     "kint-php/kint": "^4.1",
     "friendsofphp/php-cs-fixer": "~3.12.0",
     "phpunit/phpunit": "~9.5.0",
-    "brianium/paratest": "^6.2"
+    "brianium/paratest": "^6.2",
+    "guzzlehttp/guzzle": "^7.5"
```

Comment:
I'm wondering whether this isn't a good candidate for Composer suggestion: https://getcomposer.org/doc/04-schema.md#suggest